include/SFML/System/Clock.hpp

```diff
@@ -30,10 +30,34 @@
 ////////////////////////////////////////////////////////////
 #include <SFML/System/Export.hpp>
 #include <SFML/System/Time.hpp>
+#include <chrono>
+#include <ratio>
+#include <type_traits>
+
+#ifdef SFML_SYSTEM_ANDROID
+#include <SFML/System/Android/SuspendAwareClock.hpp>
+#endif
 
 
 namespace sf
 {
+namespace priv
+{
+
+struct MostSuitableClockDetector
+{
+#if defined(SFML_SYSTEM_ANDROID) && defined(SFML_ANDROID_USE_SUSPEND_AWARE_CLOCK)
+    using type = SuspendAwareClock;
+#else
+    using type = std::conditional_t<
+        std::chrono::high_resolution_clock::is_steady,
+        std::chrono::high_resolution_clock,
+        std::chrono::steady_clock>;
```

Comment:
Could you maybe write a comment (and if it helps, a link) as to why this is done? Would be nice to preserve your valuable research.

Reply:
Added documentation to the class